src/Service/VendorService/RbDigital/RbDigitalMagazinesVendorService.php

```diff
@@ -0,0 +1,176 @@
+    {
+        parent::__construct($eventDispatcher, $entityManager, $statsLogger);
+
+        $this->searchService = $searchService;
+        $this->httpClient = $httpClient;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function load(bool $queue = true, int $limit = null): VendorImportResultMessage
+    {
+        if (!$this->acquireLock()) {
+            return VendorImportResultMessage::error(parent::ERROR_RUNNING);
+        }
+
+        $this->queue = $queue;
+        $this->progressStart('Search data well for: "'.self::VENDOR_SEARCH_TERM.'"');
+
+        $offset = 1;
+        try {
+            do {
+                $this->progressMessage('Search data well for: "'.self::VENDOR_SEARCH_TERM.'" (Offset: '.$offset.')');
+
+                // Search the data well for material with acSource set to "comics plus".
+                [$pidArray, $more, $offset] = $this->searchService->search(self::VENDOR_SEARCH_TERM, $offset);
+
+                // Get the data wells image urls from the results
+                $pidArray = $this->getImageUrls($pidArray);
+
+                // Get the redirect target for the image urls
+                $pidArray = $this->getRedirectedUls($pidArray, $offset - SearchService::SEARCH_LIMIT);
+
+                // Remove 'format' parameter from image urls
+                RbDigitalMagazinesPublicUrlConverter::convertArrayValues($pidArray);
+
+                $batchSize = \count($pidArray);
+                $this->updateOrInsertMaterials($pidArray, IdentifierType::PID, $batchSize);
+
+                $this->progressMessageFormatted($this->totalUpdated, $this->totalInserted, $this->totalIsIdentifiers);
+                $this->progressAdvance();
+
+                if ($limit && $this->totalIsIdentifiers >= $limit) {
+                    $more = false;
+                }
+            } while ($more);
+
+            return VendorImportResultMessage::success($this->totalIsIdentifiers, $this->totalUpdated, $this->totalInserted, $this->totalDeleted);
+        } catch (\Exception $exception) {
+            return VendorImportResultMessage::error($exception->getMessage());
+        }
+    }
+
+    /**
+     * Get the image urls for all results in array.
+     *
+     * @param array $pidArray
+     *   An array of pid => result to be converted
+     *
+     * @return array
+     */
+    private function getImageUrls(array $pidArray): array
+    {
+        $result = [];
+        foreach ($pidArray as $pid => $item) {
+            $result[$pid] = $this->getImageUrl($item['record']['identifier']);
+        }
+
+        return $result;
+    }
+
+    /**
+     * Get the image urls from the data well result.
+     *
+     * @param array $result
+     *   A data well result array
+     *
+     * @return string|null
+     */
+    private function getImageUrl(array $result): ?string
```

Comment:
Perhaps change the function name to something more distinct from "getImageUrls()" above